proxy/http2/Http2ConnectionState.cc

```diff
@@ -387,6 +387,12 @@ rcv_priority_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
                       "priority parse error");
   }
 
+  //  A stream cannot depend on itself.  An endpoint MUST treat this as a stream error of type PROTOCOL_ERROR.
+  if (stream_id == priority.stream_dependency) {
+    return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_CONNECTION, Http2ErrorCode::HTTP2_ERROR_PROTOCOL_ERROR,
```

Comment:
The spec says treat it as a stream error, so please use `HTTP2_ERROR_CLASS_STREAM` instead.